codecov/gazebo
# ref: Convert useRepo to TS

# Description

Converts the useRepos hook to TS and updates the respective test mocks accordingly

Used the graphQL schemas from codecov-api as a guide.
https://github.com/codecov/codecov-api/blob/06b40b804131fef5a6962522d55f0e4baf1977b4/graphql_api/types/owner/owner.graphql and https://github.com/codecov/codecov-api/blob/6be2a744f32168bd621f375164c0073671d6f340/graphql_api/types/repository/repository.graphql

# Manual Testing

- Ran through the app clicking around to confirm there were no side effects from improper zod schema type casting
- Confirmed UTs passed after mock updates

# Link to Sample Entry



# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.
